packages/one-app-bundler/webpack/loaders/common.js

```diff
@@ -39,30 +39,73 @@ const cssLoader = ({ name = '', importLoaders = 2 } = {}) => ({
 
 /* eslint-disable inclusive-language/use-inclusive-words --
 config options for a third party library */
-const purgeCssLoader = () => {
-  const configOptions = getConfigOptions();
-  const whitelistPatterns = configOptions.purgecss.whitelistPatterns
-    ? configOptions.purgecss.whitelistPatterns.map((pattern) => new RegExp(pattern, 'i'))
+// transform strings deep and greedy parameters into regex
+const reconcileSafeList = (safelist) => {
+  const configOptionsReconciled = safelist;
+
+  const greedy = safelist.greedy
+    ? safelist.greedy.map((pattern) => new RegExp(pattern, 'i'))
     : [];
-  const whitelistPatternsChildren = configOptions.purgecss.whitelistPatternsChildren
-    ? configOptions.purgecss.whitelistPatternsChildren.map((pattern) => new RegExp(pattern, 'i'))
+  configOptionsReconciled.greedy = greedy;
+  const deep = safelist.deep
+    ? safelist.deep.map((pattern) => new RegExp(pattern, 'i'))
     : [/:global$/];
-  if (configOptions.purgecss.disabled) return [];
+  configOptionsReconciled.deep = deep;
```

Comment:
just fyi, `safelist` is a pointer, this means that `safelist` points to the provided object and `configOptionsReconciled` points to `safelist` because of `const configOptionsReconciled = safelist;`, any mutation performed on `configOptionsReconciled` ends up mutating the provided object.

Here's an example

```js
const reconcileSafeList = (safelist) => {
  const configOptionsReconciled = safelist;

  configOptionsReconciled.b = 2

  console.log('--configOptionsReconciled', JSON.stringify(configOptionsReconciled))
}

const test = {
  a: 1
};

console.log('--test', JSON.stringify(test))

reconcileSafeList(test)

console.log('--test', JSON.stringify(test))
```

I would, instead, do a deep clone of `safelist` so we don't mutate the original object